src/main/java/com/boot/service/ItemService.java

```diff
@@ -104,19 +99,10 @@
             return ResponseEntity.badRequest().body(new MessageResponse("The basket is empty!"));
         }
 
-        //get full information about item
-        List<Item> namesOfUsersItems = userItemList.stream().map(item -> item.getItem()).collect(Collectors.toList());
-        List<String> fullInformationAboutItem = new ArrayList<>();
-        for (Item usersItem : namesOfUsersItems) {
-            fullInformationAboutItem.add(itemRepository.findAllByName(usersItem.getName()).getName());
-            fullInformationAboutItem.add(itemRepository.findAllByName(usersItem.getName()).getDescription());
-        }
-
-        // send inf about item on the users email
-        String message = "the products you bought: " + fullInformationAboutItem;
-        // fullInformationAboutItem.stream().forEach(item -> System.out.println(item));
-        mailSender.send(userEmail, "your list of purchased items ", message);
-        //clean the basket
+        //get a list of all the user's items and send information about each item to the user's email
+        List<Item> userItems = userItemList.stream().map(item -> item.getItem()).collect(Collectors.toList());
```

Comment:
Where userItemList came from ? userItemList might be null or include nulls. In both cases you might have exception at line 103 and you wont be able to find reason what ever it is NPE in map function or stream called on null list